scripts/tests/test_mount_write_encrypted.sh

```
@@ -0,0 +1,87 @@
+  if [ -n "${MOUNT_PID:-}" ]; then
+    kill "$MOUNT_PID" 2>/dev/null || true
+  fi
+  if mount | grep -q "$MOUNTPOINT"; then
+    fusermount -u "$MOUNTPOINT" 2>/dev/null || umount "$MOUNTPOINT" 2>/dev/null || true
+  fi
+  rm -rf "$WORKDIR"
+}
+trap cleanup EXIT
+
+# Create encrypted archive with one file (use relative path for predictable entry name)
+(cd "$WORKDIR" && echo "original" > seed.txt && \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)